### PR TITLE
Added NPQ test functions

### PR DESCRIPTION
**Pull request by MOchiara**

### CODE REVIEW

**Comment by MOchiara** [comment]
@callumrollo

File: glidertest/tools.py [hunk 1]
```diff
@@ -159,3 +163,138 @@ def chl_first_check(ds):
...
+
+    # reindex days to original dataframe as night
+    df_reidx = grp_avg.reindex(df.index)
+    sunrise, sunset = df_reidx[["sunrise", "sunset"]].values.T
+
+    return sunrise, sunset
+
+
+def test_npq(ds, offset=np.timedelta64(1, "h"), start_time='2024-04-18', end_time='2024-04-20', sel_day=6):
+    ds = ds.set_xindex('TIME')
```
Comment (callumrollo) [inline glidertest/tools.py:267]: docstring here too please

File: notebooks/demo.ipynb [hunk 2]
```diff
@@ -98,7 +99,17 @@
   {
    "cell_type": "code",
```
Comment (callumrollo) [inline notebooks/demo.ipynb:100]: please add a markdown cell here explaining what this does

**Review by callumrollo — CHANGES_REQUESTED** [review]

File: glidertest/tools.py [hunk 3]
```diff
@@ -159,3 +163,138 @@ def chl_first_check(ds):
...
+    df = DataFrame.from_dict(dict([("time", time), ("lat", lat), ("lon", lon)]))
+
+    # set days as index
+    df = df.set_index(df.time.values.astype("datetime64[D]"))
+
+    # groupby days and find sunrise for unique days
+    # groupby days and find sunrise/sunset for unique days
+    grp_avg = df.groupby(df.index).mean(numeric_only=False)
+    date = grp_avg.index.to_pydatetime()
+    date = grp_avg.index
```
[the remark this answers is not in the document]
Comment (MOchiara) [reply glidertest/tools.py:181]: this is all glider tools and I haven't modified it ahah

[the remark this answers is not in the document]
Comment (callumrollo) [reply glidertest/tools.py:181]: then whoever wrote it should feel bad! Also you should reference in the docstring if it's a function copied from somewhere else

File: glidertest/tools.py [hunk 4]
```diff
@@ -159,3 +163,138 @@ def chl_first_check(ds):
     else:
         print(
             'Data from the deepest 10% of data has been analysed and data seems stable. These deep values can be used to re-assess the dark count if the no chlorophyll at depth assumption is valid in this site and this depth')
+def sunset_sunrise(time, lat, lon):
```
[the remark this answers is not in the document]
Comment (MOchiara) [reply glidertest/tools.py:166]: I will add it, we can later doc strings to all the functions once we say we like them

**Review by callumrollo — CHANGES_REQUESTED** [review]
Looking good! One final change: rename the function `test_npq` to something else. You only name a function starting with `test_` if it is a software test

**Comment by MOchiara** [comment]
Very true with the function starting with test, my bad. Changed now

**Review by callumrollo — APPROVED** [review]

**Pull request merged**